mkosi/backend.py

```diff
@@ -52,18 +54,54 @@ class MkosiException(Exception):
 ARG_DEBUG = ()
 
 
+class Parseable:
+    "A mix-in to provide conversions for argparse"
+
+    def __repr__(self) -> str:
+        """Return the member name without the class name"""
+        return cast(str, getattr(self, "name"))
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    @classmethod
+    def from_string(cls: Any, name: str) -> Any:
+        """A convenience method to be used with argparse"""
+        try:
+            return cls[name]
+        except KeyError:
+            raise argparse.ArgumentTypeError(f"unknown Format: {name!r}")
+
+    @classmethod
+    def parse_list(cls: Any, string: str) -> List[Any]:
+        return [cls.from_string(p) for p in string.split(",") if p]
+
+
+class PackageType(enum.Enum):
+    rpm = 1
+    deb = 2
+    pkg = 3
+    bundle = 4
+
+
 class Distribution(enum.Enum):
-    fedora = 1
-    debian = 2
-    ubuntu = 3
-    arch = 4
-    opensuse = 5
-    mageia = 6
-    centos = 7
-    centos_epel = 8
-    clear = 9
-    photon = 10
-    openmandriva = 11
+    fedora = PackageType.rpm
+    debian = PackageType.deb
+    ubuntu = PackageType.deb
+    arch = PackageType.pkg
+    opensuse = PackageType.rpm
+    mageia = PackageType.rpm
+    centos = PackageType.rpm
+    centos_epel = PackageType.rpm
+    clear = PackageType.bundle
+    photon = PackageType.rpm
+    openmandriva = PackageType.rpm
+
+    def __new__(cls, package_type: PackageType) -> "Distribution":
+        entry = object.__new__(cls)
+        entry._value_ = len(cls.__members__)
+        entry.package_type = package_type
+        return cast("Distribution", entry)
```

Comment:
That is an interesting construction. Took me a while to wrap my head around it. Why not a property on the enum looking up the type in a dictionary? Maybe a tad slower and the list of distros is doubled, but a bit less magical. :)

Reply:
No, that's understandable, but maybe add a link to the [specific doc section](https://docs.python.org/3.9/library/enum.html#when-to-use-new-vs-init) as a comment.

Maybe an additional idea: I think this would also be less magical, if you'd pass the index in explicitly instead of of calculating it dynamically via `len(cls.__members__)`, i.e. 
```python
class Distribution(enum.Enum):
    fedora = (0, PackageType.rpm)
    debian = (1, PackageType.deb)
```
While this is repetitive, I mostly stumbled over things getting the same value, because multiple distros do share packaging types. This would also be in line with the example from the docs.